### PR TITLE
k9s: add aliases, plugins, views

### PR DESCRIPTION
**Pull request by katexochen**
### Description

Adding the remaining config files for k9s. Documented at https://k9scli.io/

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@liyangau

### CODE REVIEW

File: modules/programs/k9s.nix [hunk 1]
```diff
@@ -50,6 +50,23 @@ in {
       '';
     };
 
+    aliases = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      description = ''
+        aliases written to
```
Comment (sumnerevans) [inline modules/programs/k9s.nix:57]:
```suggestion
        Aliases written to
```

File: modules/programs/k9s.nix [hunk 2]
```diff
@@ -72,6 +89,66 @@ in {
         };
       '';
     };
+
+    plugin = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      description = ''
+        plugins written to
```
Comment (sumnerevans) [inline modules/programs/k9s.nix:97]:
```suggestion
        Plugins written to
```

File: modules/programs/k9s.nix [hunk 3]
```diff
@@ -72,6 +89,66 @@ in {
...
+          };
+        };
+      '';
+    };
+
+    views = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      description = ''
+        resource column views written to
```
Comment (sumnerevans) [inline modules/programs/k9s.nix:129]:
```suggestion
        Resource column views written to
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
just nits, otherwise looks good

**Comment by rycee** [comment]
Thanks for the contribution! I rebased it to master in 691cbcc03af6ad1b5384c0e0e0b5f2298f58c5ce 🙂

**Pull request closed without merging**